src/edit-plan-form.ts

```diff
@@ -3,26 +3,63 @@ import { customElement, property } from 'lit/decorators.js';
 
 import type { MonthlyPlan } from './models/plan';
 import './form-sections/cancel';
+import './form-sections/amount';
+import type { MGCEditPlanAmount } from './form-sections/amount';
 
 @customElement('ia-mgc-edit-plan')
 export class IauxEditPlanForm extends LitElement {
   @property({ type: Object }) plan?: MonthlyPlan;
 
   @property({ type: Object }) cancelPlanHandler?: (plan: MonthlyPlan) => void;
 
+  @property({ type: Object }) updateAmountHandler?: (
+    plan: MonthlyPlan,
+    amountUpdates: {
+      amount: number;
+      baseAmount: number;
+      coverFees: boolean;
+      feeCovered: number;
+    }
+  ) => void;
```

Comment:
_suggestion_ Usually it's better to respond to an event than a callback for webcomponent data shuffling. ie.. emit this data payload as an event when the amount changes and the consumer can handle it.